doc/api/worker.md

```diff
@@ -0,0 +1,343 @@
+# Worker
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+The `worker` module provides a way to create multiple environments running
+on independent threads, and to create message channels between those. It
+can be accessed using:
+
+```js
+const worker = require('worker');
+```
+
+Workers are useful for performing CPU-intensive JavaScript operations; do not
+use them for I/O, since Node’s built-in mechanisms for performing operations
+asynchronously already treat it more efficiently than Worker threads can.
+
+Workers can also, unlike child processes, share memory efficiently by
+transferring `ArrayBuffer` instances or sharing `SharedArrayBuffer` instances
+between them.
+
+## Example
+
+```js
+const { Worker, isMainThread, postMessage, workerData } = require('worker');
+
+if (isMainThread) {
+  module.exports = async function parseJSAsync(script) {
+    return new Promise((resolve, reject) => {
+      const worker = new Worker(__filename, {
+        workerData: script
+      });
+      worker.on('message', resolve);
+      worker.on('error', reject);
+      worker.on('exit', (code) => {
+        if (code !== 0)
+          reject(new Error(`Worker stopped with exit code ${code}`));
+      });
+    });
+  };
+} else {
+  const { parse } = require('some-js-parsing-library');
+  const script = workerData;
+  postMessage(parse(script));
+}
+```
+
+## Event: 'workerMessage'
+<!-- YAML
+added: REPLACEME
+-->
+
```

Comment:
What arguments does this event get? `(value, transferList)`?

Reply:
Only `value`. We could try to add the `transferList` too, if that seems useful, but all of the relevant objects should be reachable through `value` anyway.

I don’t know if we have some standard format for saying “this events has one parameter that can be any JS value”, so I’m adding prose for it.

Reply:
I think something like the following is used. Basically the same format as function parameters.

```markdown
### Event: 'event'

* `value` {any} The value

This event is fired when ...
```

Reply:
Ah, thanks – done!